java/bigtable-proxy/src/main/java/com/google/cloud/bigtable/examples/proxy/core/CallProxy.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.examples.proxy.core;
+
+import io.grpc.ClientCall;
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.Status;
+import javax.annotation.concurrent.GuardedBy;
+
+/** A per RPC proxy. */
+class CallProxy<ReqT, RespT> {
+  final RequestProxy serverCallListener;
+  final ResponseProxy clientCallListener;
+
+  public CallProxy(ServerCall<ReqT, RespT> serverCall, ClientCall<ReqT, RespT> clientCall) {
+    serverCallListener = new RequestProxy(clientCall);
+    clientCallListener = new ResponseProxy(serverCall);
+  }
+
+  private class RequestProxy extends ServerCall.Listener<ReqT> {
```

Comment:
Maybe add a brief comment explaining the flow here (or perhaps to `onMessage`). Same for Response proxy.

IIUC onMessage is called multiple times until a full request message is received (I think this is what turns `clientCall.isReady() == true`?), and then it sends that to the server. Is this right?

Reply:
done